pkgs/applications/window-managers/volare/default.nix

```diff
@@ -0,0 +1,95 @@
+{ lib, stdenv, fetchFromGitea, fetchpatch, substituteAll, swaybg, sway
+, meson, ninja, pkg-config, wayland-scanner, scdoc, python3
+, wayland, libxkbcommon, pcre2, json_c, libevdev
+, pango, cairo, libinput, gdk-pixbuf, librsvg
+, wlroots_0_16, wayland-protocols, libdrm
+, nixosTests
+# Used by the NixOS module:
+, isNixOS ? false
+, enableXWayland ? true, xorg
+, systemdSupport ? lib.meta.availableOn stdenv.hostPlatform systemd, systemd
+, trayEnabled ? systemdSupport
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "volare-unwrapped";
+  version = "unstable-2023-07-06";
+
+  inherit enableXWayland isNixOS systemdSupport trayEnabled;
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "raboof";
+    repo = "volare";
+    rev = "040c7b93a99f603d820936f0f46e4b9c54181a62";
+    hash = "sha256-R2HcpKwS8SgVsSkJRsqoj3b4wYpIQHxSxsSctGF3qqE=";
+  };
+
+  patches = [
+    ./load-configuration-from-etc.patch
+
+    (substituteAll {
+      src = ../sway/fix-paths.patch;
+      inherit swaybg;
+    })
+  ] ++ lib.optionals (!finalAttrs.isNixOS) [
+    # References to /nix/store/... will get GC'ed which causes problems when
+    # copying the default configuration - TODO adapt to volare
+    #../sway/sway-config-no-nix-store-references.patch
+  ] ++ lib.optionals finalAttrs.isNixOS [
+    # Use /run/current-system/sw/share and /etc instead of /nix/store
+    # references:
+    ../sway/sway-config-nixos-paths.patch
+  ];
+
+  strictDeps = true;
+  depsBuildBuild = [
+    pkg-config
+  ];
+
+  nativeBuildInputs = [
+    meson ninja pkg-config wayland-scanner scdoc
+  ];
+
+  buildInputs = [
+    wayland libxkbcommon pcre2 json_c libevdev
+    pango cairo libinput gdk-pixbuf librsvg
+    wayland-protocols libdrm
+    (wlroots_0_16.override { inherit (finalAttrs) enableXWayland; })
+    sway
+  ] ++ lib.optionals finalAttrs.enableXWayland [
+    xorg.xcbutilwm
+  ];
+
+  mesonFlags = let
+    # The "sd-bus-provider" meson option does not include a "none" option,
+    # but it is silently ignored iff "-Dtray=disabled".  We use "basu"
+    # (which is not in nixpkgs) instead of "none" to alert us if this
+    # changes: https://github.com/swaywm/sway/issues/6843#issuecomment-1047288761
+    # assert trayEnabled -> systemdSupport && dbusSupport;
+
+    sd-bus-provider =  if systemdSupport then "libsystemd" else "basu";
+    in
+    [ "-Dsd-bus-provider=${sd-bus-provider}" ]
+    ++ lib.optional (!finalAttrs.enableXWayland) "-Dxwayland=disabled"
+    ++ lib.optional (!finalAttrs.trayEnabled)    "-Dtray=disabled"
+  ;
+
+  preInstall = ''
+    sed -i "s|#\!/usr/bin/env python3|#\!${python3}/bin/python3|" ../meson_volare_rename.py
+  '';
+
+  meta = with lib; {
+    description = "A tiling, tabbed wayland compositor (based on Sway)";
```

Comment:
It appears that `volare` is a fork of `sway`?  The `longDescription` needs to explain what `volare` changes or improves.

Reply:
I'll change it to "other tiling compositors such as Sway" to make it more explicit, but the `longDescription` already explains what it changes :)